TODO/the-ultimate-guide-to-creating-a-mobile-application.md

```diff
@@ -8,191 +9,191 @@
+一个值得注意的细节是扩展性在移动应用中是很重要的东西。计划软件架构可以和设计齐头并进。这是一个复杂的过程：通常会包括许多次迭代，同时接收设计团队和开发团队的不断反馈。架构设计的主要目标是建立健全完整软件应用架构（前端和后端）。在这个阶段开发者决定最有效率的 开发和维护应用的技术和工艺解决方案。最终决定取决于诸多因素，如信息传输环境，设计的复杂性，存储数据的必要性和记录功能等。后端开发通过网络建立与应用双向同步数据的桥梁。
 
-### iOS development
+### iOS 开发
 
-The actual process of coding the app here in Tubik is done in Xcode as the effective and flexible platform with broad functionality for iOS developers. As it is defined on the [official website](http://help.apple.com/xcode/mac/8.0/#/devc8c2a6be1), «*Xcode is Apple’s integrated development environment (IDE). You use Xcode to build apps for Apple products, including iPad, iPhone, Apple Watch, Apple TV, and Mac. Xcode provides tools to manage your entire development workflow — from creating your app to testing, optimizing, and submitting it to the App Store.*»
+Tubik 上开发软件的真正过程其实是在 Xcode  这个高效灵活且功能强大的开发 iOS软件的平台上完成的。就像在 [官方网站](http://help.apple.com/xcode/mac/8.0/#/devc8c2a6be1) 上描述的那样，**Xcode 是 Apple 的集成开发环境（IDE）。使用 Xcode来为苹果设备开发软件，包括 iPad、iPhone、Apple Watch、Apple TV 和 Mac。Xcode提供管理整个工作流的工具，从建立应用到测试、优化和在 Apple Store 上架。**
 
-Xcode only runs on Macs and has everything developers need to build iOS apps. There are no official ways to launch it on a Windows or Linux PC. This means that if somebody wants to develop an iPhone, iPad, Apple Watch, Apple TV and Mac apps but doesn’t have a Mac, he or she will need to buy it.
+Xcode 只在 Mac 设备上运行，并且拥有开发 iOS 应用所需的一切。官方没有提供在 Windows 或者 Linux 上运行的方式。这意味着如果想为iPad、iPhone、Apple Watch、Apple TV 和 Mac 开发应用而没有 Mac 的话，就必须买一台去。
 
-Two basic and most popular programming languages for iOS development are Objective-C and Swift — we considered benefits and drawbacks of each in our [earlier post](http://tubikstudio.com/swift-or-objective-c-tool-tips-for-ios-developers/). Since 2014, Swift language has been used by Xcode for coding more often than Objective-C, which is a difficult language for developers to learn and code. Still, Objective-C is a good fit for early iPhone hardware which has less RAM and slower processes and it is also effectively applicable for maintaining and updating the apps initially created in it. Swift keeps step with modern hardware and it is also faster in the process of coding, it is more easy-to-use, shorter and secure. For creating iOS 10 apps, Xcode 8 and the iOS 10 SDK need to be installed, also available from Apple. There are significant changes to the Swift language and the SDK in this version of Xcode.
+iOS 开发者最常用的两种语言是 Objective-C 和 Swift ——我们在 [早期的博文](http://tubikstudio.com/swift-or-objective-c-tool-tips-for-ios-developers/) 里对比了两者的优缺点。从2014年开始，在 Xcode 中 Swift 就超过了 Objective-C，这对于开发者是个难以上手的语言。尽管如此，Objective-C 对于早期低 RAM 和处理器的 iPhone 硬件来说有着良好适配性，并且方便先前用 Objective-C 应用的更新和维护。Swift 紧跟最新设备的步伐，敲代码更快，更易用，更短而且更有安全性。创建 iOS 10 的应用时，需要安装 Xcode 8 和 iOS 10 SDK，也可以在 Apple 找到。在这个版本的 Xcode 上 Swift 和 SDK 有些重要更新。
```

Comment:
Since 2014, Swift language has been used by Xcode for coding more often than Objective-C, which is a difficult language for developers to learn and code.

从2014年开始，Xcode 中 Swift 的使用率就超过了相对而言更难上手、更难编程的 Objective-C 。